app/controllers/posts_controller.rb

```diff
@@ -4,6 +4,10 @@ def index
     @posts = Post.all
     @user = User.find(params[:user_id])
     @posts = @user.posts
+    respond_to do |format|
+      format.html
+      format.json { render json: @posts }
+    end
   end
```

Comment:
- [ ] The API endpoint to list all the posts created by a user isn't functioning as expected due to Devise authentication(see image). Kindly check and make sure the response is a JSON.
![Screenshot from 2023-10-12 19-25-15](https://github.com/RileyManda/blog-app/assets/91142432/5f0e2e0a-d425-44e0-87e6-94f0399e1396)